### PR TITLE
Add 'Open File' to change list side bar context menu

### PR DESCRIPTION
**Pull request by shimpossible**
Add context menus for opening items with configurable apps
supports #2009

### CODE REVIEW

File: app/src/ui/changes/changed-file.tsx [hunk 1]
```diff
@@ -112,7 +117,13 @@ export class ChangedFile extends React.Component<IChangedFileProps, void> {
         action: () => this.props.onRevealInFileManager(this.props.path),
         enabled: this.props.status !== AppFileStatus.Deleted,
       },
+      {
+        label: 'Open file',
```
Comment (joshaber) [inline app/src/ui/changes/changed-file.tsx]: On Mac this should be Title Case:

```ts
label: __DARWIN__ ? 'Open File' : 'Open file'
```

Comment (joshaber) [inline app/src/ui/changes/changed-file.tsx]: Let's also make this a little clearer by labeling it as "Open with External Editor."

File: app/src/ui/changes/changes-list.tsx [hunk 2]
```diff
@@ -31,6 +31,11 @@ interface IChangesListProps {
    * @param path The path of the file relative to the root of the repository
    */
   readonly onRevealInFileManager: (path: string) => void
+  /**
```
Comment (joshaber) [inline app/src/ui/changes/changes-list.tsx]: Style nit: could you put an empty line before the start of this comment?

File: app/src/ui/changes/sidebar.tsx [hunk 3]
```diff
@@ -13,7 +14,7 @@ import { ICommitMessage } from '../../lib/app-state'
 import { ClickSource } from '../list'
 import { WorkingDirectoryFileChange } from '../../models/status'
 import { CSSTransitionGroup } from 'react-transition-group'
-
+import { openFile } from '../../lib/open-file'
```
Comment (joshaber) [inline app/src/ui/changes/sidebar.tsx]: Style nit: could you insert an empty line after this import?

File: app/src/ui/repositories-list/index.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { Dispatcher } from '../../lib/dispatcher'
 import { FilterList } from '../lib/filter-list'
 import { assertNever } from '../../lib/fatal-error'
 import { FoldoutType } from '../../lib/app-state'
-
+import { Repository } from '../../models/repository'
```
Comment (joshaber) [inline app/src/ui/repositories-list/index.tsx]: Style nit: could you insert an empty line after this import?

File: app/src/ui/repositories-list/index.tsx [hunk 5]
```diff
@@ -31,9 +31,16 @@ export class RepositoriesList extends React.Component<IRepositoriesListProps, vo
       key={repository.id}
       repository={repository}
       onRemoveRepository={this.props.onRemoveRepository}
+      onRevealInFileManager={this.onRevealInFileManager}
     />
   }
 
+  private onRevealInFileManager(repository: Repositoryish) {
```
Comment (joshaber) [inline app/src/ui/repositories-list/index.tsx]: This work's already been done in https://github.com/desktop/desktop/pull/2008.

File: app/src/ui/changes/changes-list.tsx (hunk 2 above)
Comment (shimpossible) [inline app/src/ui/changes/changes-list.tsx]: At the start of each comment?

`  readonly onDiscardChanges: (path: string) => void

  /**
   * Called to reveal a file in the native file manager.
   * @param path The path of the file relative to the root of the repository
   */

  readonly onRevealInFileManager: (path: string) => void
  /**
   * Called to open a file it its default application
   * @param path The path of the file relative to the root of the repository
   */
  readonly onOpenItem: (path: string) => void`

Comment (joshaber) [inline app/src/ui/changes/changes-list.tsx]: Yup!

File: app/src/ui/changes/changed-file.tsx [hunk 6]
```diff
@@ -21,6 +21,11 @@ interface IChangedFileProps {
    * @param path The path of the file relative to the root of the repository
    */
   readonly onRevealInFileManager: (path: string) => void
+  /**
```
Comment (joshaber) [inline app/src/ui/changes/changed-file.tsx]: Style nit: could you add a newline above this comment too?

**Comment by shimpossible** [comment]
:100:

**Comment by joshaber** [comment]
Awesome, thank you @shimpossible!

**Pull request merged**